src/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Tables.cs

```diff
@@ -302,15 +306,18 @@ public void SetCapacity(TableIndex table, int capacity)
             });
         }
 
-        public void AddInterfaceImplementation(
+        public InterfaceImplementationHandle AddInterfaceImplementation(
             TypeDefinitionHandle type,
             EntityHandle implementedInterface)
         {
             _interfaceImplTable.Add(new InterfaceImplRow
             {
                 Class = (uint)MetadataTokens.GetRowNumber(type),
-                Interface = (uint)CodedIndex.ToTypeDefOrRef(implementedInterface)
+                Interface = (uint)CodedIndex.ToTypeDefOrRefOrSpec(implementedInterface)
             });
+
+            // TODO:
+            return (InterfaceImplementationHandle)MetadataTokens.Handle(TableIndex.InterfaceImpl, _interfaceImplTable.Count);
```

Comment:
Yes, it can. I still have a copy of the code in Roslyn where it can't be fixed, so I'm keeping it in sync for now. I hope to get rid of the copy soon, after all the major API refactorings are done. I'm then gonna follow up with TODO cleanup (i keep a list).